### PR TITLE
Cosmetic re-arrangements to miner state and shared code.

### PR DESCRIPTION
**Pull request by anorth**
Only minor semantic changes, like replacing some assertions with errors.

This is a prefactor to finishing of the power actor data and structure.

### CODE REVIEW

File: actors/builtin/storage_power/storage_power_actor_state.go [hunk 1]
```diff
@@ -129,63 +118,63 @@ func (st *StoragePowerActorState) _selectMinersToSurprise(s adt.Store, challenge
...
+func (st *StoragePowerActorState) deductClaimedPowerForSector(s adt.Store, minerAddr addr.Address, weight autil.SectorStorageWeightDesc) error {
+	sectorPower := indices.ConsensusPowerForStorageWeight(weight)
 
 	currentPower, ok, err := getStoragePower(s, st.ClaimedPower, minerAddr)
 	if err != nil {
 		return errors.Wrap(err, "failed to get claimed miner power")
 	}
-	Assert(ok)
+	if !ok {
+		return errors.Errorf("no power for actor %v", minerAddr)
```
Comment (frrist) [inline actors/builtin/storage_power/storage_power_actor_state.go]: for here and below is it worth stating that this is "Claimed" power?

File: actors/builtin/storage_market/storage_market_actor_state.go [hunk 2]
```diff
@@ -15,16 +15,17 @@ import (
 const epochUndefined = abi.ChainEpoch(-1)
 
 // TODO AMT
```
[the remark this answers is not in the document]
Comment (anorth) [reply actors/builtin/storage_market/storage_market_actor_state.go]: I'm not sure, deferring the necessary investigation for now.

File: actors/builtin/storage_market/storage_market_actor_state.go [hunk 3]
```diff
@@ -15,16 +15,17 @@ import (
...
-type DealsAMT map[abi.DealID]OnChainDeal
+type DealsById map[abi.DealID]OnChainDeal
 
 // TODO AMT
-type CachedDealIDsByPartyHAMT map[addr.Address]actor_util.DealIDSetHAMT
+type DealsByParty map[addr.Address]DealIDSet
```
[the remark this answers is not in the document]
Comment (anorth) [reply actors/builtin/storage_market/storage_market_actor_state.go]: I'm not sure, depends whether DealIDSet works well as a bitfield. I haven't investigated the usage pattern enough to determine yet.

File: actors/builtin/storage_market/storage_market_actor_state.go [hunk 4]
```diff
@@ -15,16 +15,17 @@ import (
...
+type DealIDSet map[abi.DealID]bool // TODO: HAMT or bitfield, depending how often it mutates
 
 // TODO AMT
-type CachedExpirationsPendingHAMT map[abi.ChainEpoch]DealIDQueue
+type DealExpirationQueue map[abi.ChainEpoch]DealIDQueue
```
[the remark this answers is not in the document]
Comment (anorth) [reply actors/builtin/storage_market/storage_market_actor_state.go]: Very likely yes.

File: actors/builtin/storage_power/storage_power_actor.go [hunk 5]
```diff
@@ -20,19 +20,12 @@ import (
...
-var TODO = autil.TODO
 
 type ConsensusFaultType int
-
 const (
-	UncommittedPowerFault ConsensusFaultType = 0
+	//UncommittedPowerFault ConsensusFaultType = 0
 	DoubleForkMiningFault ConsensusFaultType = 1
 	ParentGrindingFault   ConsensusFaultType = 2
 	TimeOffsetMiningFault ConsensusFaultType = 3
```
[the remark this answers is not in the document]
Comment (anorth) [reply actors/builtin/storage_power/storage_power_actor.go]: I'm on the fence. It's pretty important for the explicit values to be stable, and not subject to accidental change later.

**Pull request merged**